src/SmartComponents/CreateImageWizard/CreateImageWizard.js

```diff
@@ -174,21 +247,29 @@ class CreateImageWizard extends Component {
             name: 'Image output',
             component: <WizardStepImageOutput
                 value={ this.state.release }
-                upload={ this.state.upload }
                 setRelease={ this.setRelease }
-                setUpload={ this.setUpload } />
+                toggleUploadDestination={ this.toggleUploadDestination }
+                uploadDestinations={ this.state.uploadDestinations } />
         };
+
         const StepUploadAWS = {
-            name: 'Upload to AWS',
+            name: 'Amazon Web Services',
             component: <WizardStepUploadAWS
-                upload={ this.state.upload }
+                uploadAWS={ this.state.uploadAWS }
                 setUploadOptions={ this.setUploadOptions }
-                errors={ this.state.uploadErrors } />
+                errors={ this.state.uploadAWSErrors } />
+        };
+
+        const StepTargetEnv = {
+            name: 'Target environment',
+            steps: [
+                ...(this.state.uploadDestinations.aws ? [ StepUploadAWS ] : []),
```

Comment:
Same here, might as well declare an array and add each step depending on what upload destinations are set (even if it's just checking one at this point). Using ternary conditions to get the list of substeps is gonna be difficult to read no?

Reply:
I think it will be kind of ugly regardless since it will end up being a series of if statements that push to an array of steps. But I will change it to that.